MatchGame/Game.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MatchGame
+{
+    public class Game : IGame
+    {
+        private readonly IDeck _gameDeck;
+        private readonly IPlayer _player1;
+        private readonly IPlayer _player2;
+        private MatchRuleEnum _matchRule;
+        private readonly Stack<Card> _pile = new();
+
+        public Game(IDeck gameDeck, IPlayer player1, IPlayer player2)
+        {
+            _gameDeck = gameDeck;
+            _player1 = player1;
+            _player2 = player2;
+        }
+
+        public MatchRuleEnum MatchRule => _matchRule; 
+
+        public ResultEnum DeclareWinner()
```

Comment:
Pass the players through params here?

Reply:
The implementation here looks like it could be more elegant. Also, would break the O/C principle if we added player3,etc. Should players be managed in a list in the game logic? (We'd need some `int Id` properties on the Player class added. And then manage the DeclareWinner logic with some LINQ?